developer/src/kmc-package/src/compiler/package-validation.ts

```diff
@@ -155,9 +155,20 @@ export class PackageValidation {
       return false;
     }
 
+    if(this.doesContentFileEqualOutputFilename(file, outputFilename)) {
+      this.callbacks.reportMessage(PackageCompilerMessages.Error_PackageMustNotContainItself({outputFilename}));
+      return false;
+    }
+
     return true;
   }
 
+  private doesContentFileEqualOutputFilename(file: KmpJsonFile.KmpJsonFileContentFile, outputFilename: string): boolean {
```

Comment:
For consistency with the other methods, maybe rename to
```suggestion
  private checkIfContentFileEqualsOutputFilename(file: KmpJsonFile.KmpJsonFileContentFile, outputFilename: string): boolean {
```

Reply:
I actually didn't use `checkIf` deliberately, because `checkIf...` also reports a message, whereas `does...` just checks without reporting. Probably would be better to refactor `checkIf...` functions to just `does...` in the future, but super lowpri.